crates/oxc_mangler/src/lib.rs

```diff
@@ -209,7 +211,7 @@ impl Mangler {
         let root_scope_id = scope_tree.root_scope_id();
         let mut frequencies = vec![SlotFrequency::default(); total_number_of_slots];
         for (symbol_id, slot) in slots.iter_enumerated() {
-            if symbol_table.get_scope_id(symbol_id) == root_scope_id {
+            if !self.options.top_level && symbol_table.get_scope_id(symbol_id) == root_scope_id {
```

Comment:
this should probably skip mangling it it's exported?

Reply:
If the exported variable has a long name and is used more than once, the file is smaller if the mangling is done. The same can be said for imported identifiers.

Reply:
would this change the name that the file importing this function needed to import though

e.g. would it change

```
export function foo()
```
and mangle the name foo?

Reply:
Ah, that's true. Made a PR with the failing tests: https://github.com/oxc-project/oxc/pull/7927

Reply:
thanks 🙏